doc/architecture/performance/index.rst

```diff
@@ -0,0 +1,38 @@
+Piccolo
+========
+
+CCF has its own performance testing tool (Piccolo) under the ``CCF/tests/perf-system`` 
```

Comment:
Do we want to change the directory name from `perf-system` to `piccolo` to help distinguish this tool from the other CCF perf infra?

Reply:
The other perf infra doesn't really have a fancy name, there's perfclient, and there's python wrappers for it under infra. If this needed a boring and descriptive name, it'd probably be GSA: generator, submitter and analyser, but Piccolo works just as well. +1 for updating the directory name.

Reply:
Should I do the name changing in this PR, or leave it as is for now and create a new PR?